src/libcollections/str.rs

```diff
@@ -613,26 +775,41 @@ impl str {
         UnicodeStr::split_whitespace(self)
     }
 
-    /// An iterator over the lines of a string, separated by `\n` or `\r\n`.
+    /// An iterator over the lines of a string, as string slices.
     ///
-    /// This does not include the empty string after a trailing newline or CRLF.
+    /// Lines are ended with either a newline (`\n`) or a carriage return with
+    /// a line feed (`\r\n`).
+    ///
+    /// The final line ending is optional.
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
-    /// let four_lines = "foo\nbar\n\r\nbaz";
-    /// let v: Vec<&str> = four_lines.lines().collect();
+    /// let text = "foo\r\nbar\n\nbaz\n";
+    /// let mut lines = text.lines();
+    ///
+    /// assert_eq!(Some("foo"), lines.next());
+    /// assert_eq!(Some("bar"), lines.next());
+    /// assert_eq!(Some(""), lines.next());
+    /// assert_eq!(Some("baz"), lines.next());
```

Comment:
Might be good to demonstrate lack of trailing empty slice if string contains a trailing newline. It is not mentioned in documentation anymore either.